base_user_role/models/role.py

```diff
@@ -112,6 +112,6 @@ def _compute_is_enabled(self):
     @api.multi
     def unlink(self):
         users = self.mapped("user_id")
-        res = super(ResUsersRoleLine, self).unlink()
-        users.set_groups_from_roles(force=True)
+        res = super().unlink()
+        users.filtered(lambda x: x.role_line_ids).set_groups_from_roles(force=True)
```

Comment:
Sorry for this late review...

The `force=True` parameter has been added on purpose (see the comments on this PR: https://github.com/OCA/server-tools/pull/922 and the docstring of the `set_groups_from_roles` method).
Now, a thing that could be done is to make this configuration through a `ir.config_parameter` here to tune the value of the `force` parameter, we keep the default behaviour if the parameter is set (or unset) then wo do not remove groups when we remove the last role(s).

What do you think?